### PR TITLE
Fix deprecations, enable lint warnings, and err on warnings

### PR DESCRIPTION
**Pull request by Ao-senXiong**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -32,6 +32,10 @@ shadowJar {
     }
 }
 
+tasks.compileJava {
+    options.compilerArgs += '-Xlint:deprecation'
+}
+
 spotless {
     java {
```
Comment (wmdietl) [inline build.gradle:40]: In a separate PR, can you turn on spotless formatting of gradle files?

Comment (Ao-senXiong) [reply build.gradle:40]: I am sorry I did not get it, do you mean apply spotless formatting to gradle files when compileJava it?

**Comment by Ao-senXiong** [comment]
> > Thanks! Can you also turn warnings into errors? At the moment it is easy to miss that there is a warning in the CI build.
> 
> As there are deprecation warning currently, please prepare a separate PR that fixes these warnings.

Yes, now I updated it as error and working on fixing it.

File: build.gradle (hunk 1 above)
Comment (wmdietl) [reply build.gradle:40]: No, I meant adding a gradle formatting block to spotless, similar to this: https://github.com/eisop/checker-framework/blob/388fdb094942de42e871a96171fadb80c4d1a0e1/build.gradle#L309

Comment (Ao-senXiong) [reply build.gradle:40]: Done! Please see #220 :)

File: build.gradle [hunk 2]
```diff
@@ -32,6 +32,10 @@ shadowJar {
     }
 }
 
+tasks.compileJava {
+    options.compilerArgs += ['-Xlint:deprecation', '-Werror']
```
Comment (wmdietl) [inline build.gradle:36]:
```suggestion
    options.compilerArgs += ['-Xlint', '-Werror']
```

Comment (wmdietl) [reply build.gradle:36]: Let's enable all lint warnings.

**Review by wmdietl — APPROVED** [review]
Thanks!

File: build.gradle (hunk 2 above)
Comment (Ao-senXiong) [reply build.gradle]: Thanks.

**Comment by wmdietl** [comment]
@Ao-senXiong Ping about the PR to fix the deprecation warnings.

**Comment by Ao-senXiong** [comment]
> @Ao-senXiong Ping about the PR to fix the deprecation warnings.

Thanks, already made the changes. However, the test seems only check build but not the functionality.

**Comment by wmdietl** [comment]
> Thanks, already made the changes. However, the test seems only check build but not the functionality.

@Ao-senXiong Thanks! The changes look good. Did you manually verify that hover notifications still work? Do you see a way how to automate testing?

**Review by wmdietl — APPROVED** [review]

**Comment by Ao-senXiong** [comment]
> > Thanks, already made the changes. However, the test seems only check build but not the functionality.
> 
> @Ao-senXiong Thanks! The changes look good. Did you manually verify that hover notifications still work? Do you see a way how to automate testing?

I have manually verified it still works. I think I can write some unit tests for the function.

**Pull request merged**